osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs

```diff
@@ -151,6 +146,8 @@ protected override void Selected()
 
         private void updateBeatmapDifficulties()
         {
+            if (Item == null) return;
```

Comment:
This file really confuses me now. Some places null-check, some places null-ignore, some places `Debug.Assert` for no real rhyme or reason. Pick one and stick to it consistently.

Reply:
right. my bad will adress

Reply:
i think this should be more legible now?. I did not alter existing cases of null check since they are mostly used for early returns where it being possibly null is expected from what i understand, and added asserts where the expectation is that it shouldn't be null. and thanks for the patience, not too well acquainted with this part of the language yet